perun/view/scatter_plot/run.py

```diff
@@ -0,0 +1,115 @@
+"""Scatter plot interpretation of the profile"""
+
+import click
+
+import perun.utils.cli_helpers as cli_helpers
+import perun.utils.bokeh_helpers as bokeh_helpers
+import perun.view.scatter_plot.factory as scatter_factory
+
+from perun.utils.helpers import pass_profile
+
+__author__ = 'Jiri Pavela'
+
+
+def process_title(ctx, _, value):
+    """ Creates default title for scatter plot graph, if not provided by the user.
+
+    If the value supplied from CLI is non-None, it is returned as it is. Otherwise, we try to
+    create some optimal name for the graph ourselves. We do this according to already processed
+    parameters as follows:
+
+      Plot of 'of-key' per 'per-key'
+
+    The title will be further expanded by relevant data to allow for easier identification of the
+    result, as the scatter plot might produce more than one graph.
+
+    Arguments:
+        ctx(click.Context): called context of the process
+        value(object): value that is being processed ad add to parameter
+
+    Returns:
+        object: either value (if it is non-None) or default title of the graph
+    """
+    if not value:
+        # Construct default title of the graph
+        return "Plot of '{}' per '{}'".format(ctx.params['of_key'], ctx.params['per_key'])
+    return value
+
+
+@click.command()
+@click.option('--of', '-o', 'of_key', default='amount', nargs=1,
+              show_default=True, type=str,
+              is_eager=True, callback=cli_helpers.process_resource_key_param,
+              help="Data source for the scatter plot, i.e. what will be displayed on Y axis.")
+@click.option('--per', '-p', 'per_key', default='structure-unit-size', nargs=1,
```

Comment:
Currently, the default of "structure-unit-size" is great since currently we mostly use the scatter plot for dependencies between structure size and time. We should think of future, though, where the "safest" default would be snapshot or I am thinking of something called "order" (or "class"), which would be metric-independent (maybe I told you about it---I was thinking about simply ordering the values, and clustering the near points to same class).

Reply:
Agreed, will be possibly addressed in the planned profile format refactoring?

Reply:
Yes, noted.